src/lesson5-homework.ts

```diff
@@ -0,0 +1,455 @@
+    // space property
+    space: number;
+    // install method
+    install(): void;
+  }
+
+  // TODO: declare class Widget which implements IWidget
+  class Widget implements IWidget, IWidgetPrintable {
+    constructor(public name: string) {}
+    toString(): void {
+      console.log(`Widget - name: ${this.name}.`);
+    }
+  }
+
+  // TODO: declare class WidgetWithSize which implements IWidgetWithSize
+  class WidgetWithSize implements IWidgetWithSize, IWidgetPrintable {
+    constructor(
+      public name: string,
+      public width: number,
+      public height: number,
+      public color: string
+    ) {}
+    resize(): void {}
+    toString(): void {
+      console.log(
+        `Widget with size - name: ${this.name}; width: ${this.width}; height: ${this.height}; color: ${this.color}.`
+      );
+    }
+  }
+
+  // TODO: declare class DesktopWidget which implements IDesktopWidget
+  class DesktopWidget
+    extends WidgetWithSize
+    implements IDesktopWidget, IWidgetPrintable
+  {
+    constructor(
+      public name: string,
+      public width: number,
+      public height: number,
+      public color: string,
+      public os: string
+    ) {
+      super(name, width, height, color);
+    }
+    open(): void {}
+    toString(): void {
+      console.log(
+        `Desktop widget - name: ${this.name}; width: ${this.width}; height: ${this.height}; color: ${this.color}; os: ${this.os}.`
+      );
+    }
+  }
+
+  // TODO: declare class MobileWidget which implements IMobileWidget
+  class MobileWidget
+    extends WidgetWithSize
+    implements IMobileWidget, IWidgetPrintable
+  {
+    constructor(
+      public name: string,
+      public width: number,
+      public height: number,
+      public color: string,
+      public space: number
+    ) {
+      super(name, width, height, color);
+    }
+    install(): void {}
+    toString(): void {
+      console.log(
+        `Mobile widget - name: ${this.name}; width: ${this.width}; height: ${this.height}; color: ${this.color}; space: ${this.space}.`
+      );
+    }
+  }
+
+  // TODO: declare class DesktopAndMobileWidget which implements IDesktopWidget and IMobileWidget
+  class DesktopAndMobileWidget
+    extends DesktopWidget
+    implements IDesktopWidget, IMobileWidget, IWidgetPrintable
+  {
+    constructor(
+      public name: string,
+      public width: number,
+      public height: number,
+      public color: string,
+      public os: string,
+      public space: number
+    ) {
+      super(name, width, height, color, os);
+    }
+    install(): void {}
+    toString(): void {
+      console.log(
+        `Desktop and mobile widget - name: ${this.name}; width: ${this.width}; height: ${this.height}; color: ${this.color}; os:${this.os}; space: ${this.space}.`
+      );
+    }
+  }
+
+  // TODO: declare interface IWidgetPrintable wich has toString method
+  interface IWidgetPrintable {
+    // toString method - returns string
+    toString(): void;
```

Comment:
this needs to return string type

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/toString
`
class Dog {
  constructor(name, breed, color, sex) {
    this.name = name;
    this.breed = breed;
    this.color = color;
    this.sex = sex;
  }
  toString() {
    return `Dog ${this.name} is a ${this.sex} ${this.color} ${this.breed}`;
  }
}

const theDog = new Dog("Gabby", "Lab", "chocolate", "female");
console.log(`${theDog}`); // 
`